AssemblerLib/VectorMatrixAssembler.h

```diff
@@ -49,15 +56,46 @@ class VectorMatrixAssembler
     {
         assert(_data_pos.size() > id);
 
+        std::vector<double> localX;
+        std::vector<double> localX_pts;
         LocalToGlobalIndexMap::RowColumnIndices const& indices = _data_pos[id];
+        LocalToGlobalIndexMap::LineIndex remapped_rows_cols;
+
+        const unsigned element_dof = indices.rows.size();
+        auto const& x = *_x;
+        auto const& x_pts = *_x_prev_ts;
+
+        const unsigned num_comp = _data_pos.getNumComponents();
+
+        if (_x)         localX.reserve(element_dof);
+        if (_x_prev_ts) localX_pts.reserve(element_dof);
+        remapped_rows_cols.reserve(element_dof);
+
+        // The local matrix will always be ordered by component,
```

Comment:
Yes, vector might be more appropriate, here. But also the local matrix has to be ordered by component due to the `addToGlobal()` call below.